### PR TITLE
chore: update docs to accurate node support

### PR DESCRIPTION
**Pull request by hkang1**
## Description

We don't support node 17 or higher yet due to our web package dependencies.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "1.0.0",
   "private": true,
   "engines": {
-    "node": ">=16.13",
+    "node": ">=16.13 <17",
```
Comment (hamidzr) [inline webui/react/package.json:6]: I don't really know if this is the correct syntax or not but the idea lgtm :)

**Review by hamidzr — APPROVED** [review]

**Comment by hamidzr** [comment]
if this isn't strict enough we could add a version check to our makefile to stop going forward with nodes of other versions


```
diff --git a/webui/react/Makefile b/webui/react/Makefile
index 479ccf4b2..b19f6adf6 100644
--- a/webui/react/Makefile
+++ b/webui/react/Makefile
@@ -8,7 +8,7 @@ default: get-deps
 	$(MAKE) build
 
 
-.PHONY: get-deps-npm
+.PHONY: get-deps-npm check-requirements
 get-deps-npm:
 	npm ci --no-audit
 
@@ -21,7 +21,7 @@ get-deps: node_modules/done.stamp
 	git submodule update --init
 
 .PHONY: build
-build: build/done.stamp
+build: build/done.stamp check-requirements
 
 build/done.stamp: $(source_files) node_modules/done.stamp tsconfig.json craco.config.js jest.config.js
 	npm run build
@@ -48,6 +48,9 @@ check-css:
 .PHONY: check-package-lock
 check-package-lock:
 	if grep 'ssh://' package-lock.json ; then echo "ssh url in package-lock.json, please convert to https url" ; false ; fi
+.PHONY: check-requirements
+check-requirements:
+	node --version | grep -q 'v16' || (echo "node version 16.* is required" ; false)
 .PHONY: check
 check: check-js check-css check-package-lock
 
```

**Comment by hkang1** [comment]
> if this isn't strict enough we could add a version check to our makefile to stop going forward with nodes of other versions
> 
> ```
> diff --git a/webui/react/Makefile b/webui/react/Makefile
> index 479ccf4b2..b19f6adf6 100644
> --- a/webui/react/Makefile
> +++ b/webui/react/Makefile
> @@ -8,7 +8,7 @@ default: get-deps
>  	$(MAKE) build
>  
>  
> -.PHONY: get-deps-npm
> +.PHONY: get-deps-npm check-requirements
>  get-deps-npm:
>  	npm ci --no-audit
>  
> @@ -21,7 +21,7 @@ get-deps: node_modules/done.stamp
>  	git submodule update --init
>  
>  .PHONY: build
> -build: build/done.stamp
> +build: build/done.stamp check-requirements
>  
>  build/done.stamp: $(source_files) node_modules/done.stamp tsconfig.json craco.config.js jest.config.js
>  	npm run build
> @@ -48,6 +48,9 @@ check-css:
>  .PHONY: check-package-lock
>  check-package-lock:
>  	if grep 'ssh://' package-lock.json ; then echo "ssh url in package-lock.json, please convert to https url" ; false ; fi
> +.PHONY: check-requirements
> +check-requirements:
> +	node --version | grep -q 'v16' || (echo "node version 16.* is required" ; false)
>  .PHONY: check
>  check: check-js check-css check-package-lock
>  
> ```

🙏  That would save a lot of future folks from headaches. I'll add the changes.

**Comment by hamidzr** [comment]
sort of related https://github.com/determined-ai/determined/pull/3932

File: webui/react/package.json (hunk 1 above)
Comment (hkang1) [reply webui/react/package.json:6]: I wasn't sure either so I looked this up

**Comment by hkang1** [comment]
> > 🙏 That would save a lot of future folks from headaches. I'll add the changes.
> 
> I haven't tested it since I don't have nvm anymore. do you mind trying it out if you have it? thanks

gave it a thorough run with multiple `nvm` run with various versions and is working nicely

File: webui/react/Makefile [hunk 2]
```diff
@@ -48,6 +49,9 @@ check-css:
 .PHONY: check-package-lock
 check-package-lock:
 	if grep 'ssh://' package-lock.json ; then echo "ssh url in package-lock.json, please convert to https url" ; false ; fi
+.PHONY: check-requirements
+check-requirements:
+	node --version | grep -E 'v16\.1[3-6]' || (echo "node version >=16.13 is required" ; false)
```
Comment (hamidzr) [inline webui/react/Makefile:54]: maybe also print out the upper cap? <17

**Review by hamidzr — APPROVED** [review]

File: webui/react/Makefile (hunk 2 above)
Comment (hkang1) [reply webui/react/Makefile]: doh just saw this. Here's a [follow up PR](https://github.com/determined-ai/determined/pull/4290) :D

**Pull request merged**